packages/private/edtr-io/src/deserialize.ts

```diff
@@ -536,6 +536,7 @@ export function deserialize({
           content: serializeEditorState(
             toEdtr(deserializeEditorState(state.content))
           ),
+          cohesive: false,
```

Comment:
Eh, that won't work. With this, any grouped text exercise will be considered non-cohesive (even those that you saved as cohesive).

Basically: `deserialize` is responsible to get raw data from the database and deserialize it to an Edtr.io document. For this specific case, it takes the `content` fields from the text-exercise-group and respective grouped-text-exercises. The returned `type-text-exercise-group` plugin is just a fake plugin used for the Editor, it won't be persisted itself. So: you either want to persist `cohesive` somewhere else (e.g. as a new field in the entity) or move it into the `content` of text-exercise-group.

Reply:
If that brief explanation was too confusing, we can jump into a quick hangout instead.